airbyte-cdk/python/airbyte_cdk/sources/streams/utils/stream_helper.py

```diff
@@ -34,5 +34,7 @@ def get_first_record_for_slice(stream: Stream, stream_slice: Optional[Mapping[st
     :raises StopIteration: if there is no first record to return (the read_records generator is empty)
     :return: StreamData containing the first record in the slice
     """
-    records_for_slice = stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice)
+    # We wrap the return output of read_records() because some implementations return types that are iterable,
```

Comment:
Suggestion: create an issue for type checking and link it here

Reply:
Good call!
- [Issue for fixing the signature of read_records](https://github.com/airbytehq/airbyte/issues/31137)
- [Issue for enabling type-checking on the connectors](https://github.com/airbytehq/airbyte/issues/31138)